test/integration/deployment/deployment_test.go

```diff
@@ -69,3 +70,50 @@ func TestNewDeployment(t *testing.T) {
 		t.Errorf("expected new ReplicaSet last-applied annotation not copied from Deployment %s", deploy.Name)
 	}
 }
+
+// TODO(juntee): add tests for selector immutability when apps/v1beta2 infomers and listers are available
```

Comment:
@enisoc Thanks for your patient guidance last week! Some parts of the integration test are out.
I found out that integration tests are based on `extensions/v1beta1` version only. Current informers and listers support only the version too. I am not sure whether it is possible to add the integration test of `apps/v1beta2` version now, or there is a way to get over it.

I will spend some time to investigate about this, and get back to you by today.